cmd/share-download-main.go

```diff
@@ -134,8 +140,15 @@ func doShareDownloadURL(targetURL string, isRecursive bool, expiry time.Duration
 		// Recursive mode: Share list of objects
 		go func() {
 			defer close(objectsCh)
+			objectCount := 0
 			for content := range clnt.List(isRecursive, isIncomplete, DirNone) {
 				objectsCh <- content
+				objectCount ++
```

Comment:
@badscooter23 - If you can amend your commit with this fix, we can take this forward.